src/custom-elements/src/components/text-input/text-input.tsx

```diff
@@ -177,14 +181,22 @@ export class TextInput {
       ${this.iconLeft ? '-icon--left' : ''}
       ${this.iconRight ? '-icon--right' : ''}
     `;
+    const spinnerSizeMapping = {
+      sm: 'xs',
+      md: 'sm',
+      lg: 'sm--2',
+      xl: 'sm--3'
+    };
     const iconLeft = this.iconLeft && <chi-icon color={this.iconLeftColor || null} icon={this.iconLeft} />;
-    const iconRight = this.iconRight && <chi-icon color={this.iconRightColor || null} icon={this.iconRight} />;
+    const iconRight = (this.iconRight && !this.spinner) && <chi-icon color={this.iconRightColor || null} icon={this.iconRight} />;
```

Comment:
@janibolkvadze  @jllr Hiding the right icon when loading should be fine for now but we may need to revisit if search or other inputs require this icon present during a loading state.